File: Clubmanagement/src/main/java/com/example/Clubmanagement/entities/compte/generlAc/Etudiant.java
package com.example.Clubmanagement.entities.compte.generlAc;

import com.example.Clubmanagement.entities.club.Club;
import com.example.Clubmanagement.entities.compte.Clubsmembers.Members;

import javax.persistence.*;
import java.util.List;

@Entity

public class Etudiant extends Compte {



    private  Long anetud;
    private  String programme;

    //contructor


    public Etudiant() {
    }

    public Etudiant(Long anetud, String programme) {
        this.anetud = anetud;
        this.programme = programme;
    }

    public Etudiant(String civilite, String fullname, String email, String pass, Long tel, Long anetud, String programme) {
        super(civilite, fullname, email, pass, tel);
        this.anetud = anetud;
        this.programme = programme;
    }

    public Etudiant(Long id, String civilite, String fullname, String email, String pass, Long tel, Long anetud, String programme) {
        super(id, civilite, fullname, email, pass, tel);
        this.anetud = anetud;
        this.programme = programme;
    }
    //end constructors
//getters adn stters
    public Long getAnetud() {
        return anetud;
    }

    public void setAnetud(Long anetud) {
        this.anetud = anetud;
    }

    public String getProgramme() {
        return programme;
    }

    public void setProgramme(String programme) {
        this.programme = programme;
    }

//to string

    @Override
    public String toString() {
        return "Etudiant{" +
                "anetud=" + anetud +
                ", programme='" + programme + '\'' +
                '}';
    }


    public Members Jclub(){

        return null;
    }
    @ManyToMany(fetch = FetchType.LAZY, cascade = CascadeType.PERSIST)
    @JoinTable( name = "Members" , joinColumns = {@JoinColumn ( name ="Student_id",  referencedColumnName = "id", nullable = false, updatable = false)},
            inverseJoinColumns = {@JoinColumn(name = "club_id",referencedColumnName="IdClub",nullable = false, updatable = false)})
    private List<Club> C ;



}

File: Clubmanagement/src/front-end/Classes/Etudiant.ts
import {compte} from "./compte";

export class etudiant extends compte{

    private _anetud:String;
    private _progamme:String;


    get anetud(): String {
        return this._anetud;
    }

    set anetud(value: String) {
        this._anetud = value;
    }

    get progamme(): String {
        return this._progamme;
    }

    set progamme(value: String) {
        this._progamme = value;
    }

    constructor(idE: number, civilite: number, fullname: String, email: String, password: String, tel: number, anneetude: String, progamme: String) {
        super(idE, civilite, fullname, email, password, tel);
        this._anetud = anneetude;
        this._progamme = progamme;
    }

}